smarts/core/sensors.py

```diff
@@ -333,6 +334,37 @@ def _agents_alive_done_check(
 
         return False
 
+    @classmethod
+    def _actors_alive_done_check(
+        cls,
+        vehicle_index,
+        sensor_state,
+        actors_alive: Optional[ActorsAliveDoneCriteria],
+    ):
+        if actors_alive is None:
+            return False
+
+        sensor_state: SensorState = sensor_state
+        from smarts.core.vehicle_index import VehicleIndex
+
+        vehicle_index: VehicleIndex = vehicle_index
+
+        pattern = re.compile(actors_alive.actors_of_interest)
+        ## TODO optimization to get vehicles that were added and removed last step
+        ## TODO second optimization to check for already known vehicles
+        for vehicle_id in vehicle_index.vehicle_ids:
```

Comment:
When start a simulation for testing, an `TypeError` occurs here, saying that `'_CacheCallable' object is not iterable`. Maybe that's due to `vehicle_index.vehicle_ids: _CacheCallable`?